src/python/pants/build_graph/address.py

```diff
@@ -300,34 +260,14 @@ class BuildFileAddress(Address):
     :API: public
     """
 
-    def __init__(
-        self,
-        build_file: Optional[BuildFile] = None,
-        target_name: Optional[str] = None,
-        rel_path: Optional[str] = None,
-    ) -> None:
+    def __init__(self, *, rel_path: str, target_name: Optional[str] = None) -> None:
```

Comment:
This is a little not-Kosher to make it kwargs-only. But I think it's safe enough to do. 

Because `rel_path` can no longer be optional, it no longer has a default arg. So, it must come before `target_name` in the params list. This would be dangerous to reorder if people were using positional args, because they're both `str`. Instead, kwargs-only will make the error more explicit.

I also don't think this will hit many people because `build_file` was the first positional arg, and you can't give both `build_file` and `rel_path` so I suspect that most people were already using kwargs. Otherwise they would have used `BuildFileAddress(None, "target", "src/python/pants")`.

Reply:
I think this is OK, but let's hear from @stuhood ?